autoload/unite/sources/tag.vim

```diff
@@ -269,17 +269,23 @@ endfunction
 
 " filter defined by unite's parameter (e.g. Unite tag:filter)
 function! s:pre_filter(result, args)
-    if !empty(a:args)
-        let arg = a:args[0]
+    if !empty(a:args) | for arg in a:args
+        "let arg = a:args[0]
         if arg !=# ''
-            if arg ==# '/'
+            if arg ==# '%'
+            " Current buffer tags
+                let bufname = (&ft=='unite' ? expand('#:p') : expand('%:p'))
```

Comment:
`&filetype ==# 'unite'`.
And current `expand('#:p')` is bad practice.
If you want to get the buffer name, you should use `bufname(b:unite.prev_bufnr)`.